File: backend/src/main/java/com/memeit/backend/dataclasses/Poster.java
package com.memeit.backend.dataclasses;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * Created by Jv on 6/16/2018.
 */

public class Poster implements Parcelable {
    @SerializedName("pid")
    private String ID;
    @SerializedName("name")
    private String name;
    @SerializedName("pic")
    private String profileUrl;

    protected Poster(Parcel in) {
        ID = in.readString();
        name = in.readString();
        profileUrl = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(ID);
        dest.writeString(name);
        dest.writeString(profileUrl);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Poster> CREATOR = new Creator<Poster>() {
        @Override
        public Poster createFromParcel(Parcel in) {
            return new Poster(in);
        }

        @Override
        public Poster[] newArray(int size) {
            return new Poster[size];
        }
    };

    public String getID() {
        return ID;
    }

    public String getName() {
        return name;
    }

    public String getProfileUrl() {
        return profileUrl;
    }
}

File: backend/src/main/java/com/memeit/backend/dataclasses/Poster.kt
package com.memeit.backend.dataclasses

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.SerializedName

/**
 * Created by Jv on 6/16/2018.
 */

data class Poster(@SerializedName("pid")
                  val id: String?,
                  @SerializedName("name")
                  val name: String?,
                  @SerializedName("pic")
                  val profileUrl: String?) : Parcelable {


    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString())

    fun toUser(): User {
        return User(this)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(id)
        parcel.writeString(name)
        parcel.writeString(profileUrl)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Poster> {
        override fun createFromParcel(parcel: Parcel): Poster {
            return Poster(parcel)
        }

        override fun newArray(size: Int): Array<Poster?> {
            return arrayOfNulls(size)
        }
    }


}
